### DISCUSSION TITLE
Hasura cannot handle unique constraints!?

### DISCUSSION THREAD

**Discussion by some-user123** [hasura/graphql-engine#9097] [Question]
I wanted to try out Hasura on an existing project with an obviously existing database. To my understanding that is a valid and common use case. However, I could not track several of my tables, because they contain unique constraints. That behavior seems identical to this reported issue: https://github.com/hasura/graphql-engine/issues/7546 which did not receive any comment for over a year.

I tried to follow the workaround. Dropped all unique constraints, tracked the tables, created the unique constraints again. Seemes to work. Hasura even finds the unique contraints and displays them. Until you restart Hasura. Then Hasura won't even start. https://github.com/hasura/graphql-engine/issues/7850

Am I doing something fundamentally wrong? Am I the only person who wants to use Hasura with an existing database with unique constraints!? I'm a bit puzzled. I don't want Hasura to manage the schema/migrations for now... maybe later, maybe not because there is a solution already that manages db migrations.

**Comment by BenoitRanque** [comment]
Can you confirm if this is the same issue as #7850, which includes a workaround?

Can you confirm that when you re-created the unique constraints, you renamed them so they would be valid graphql identifiers?

**Reply by dsandip** [reply]
@dameleney @codedmart FYI
